src/execution/operator/aggregate/physical_hash_aggregate.cpp

```diff
@@ -45,8 +44,16 @@ PhysicalHashAggregate::PhysicalHashAggregate(ClientContext &context, vector<Logi
 		}
 
 		aggregate_return_types.push_back(aggr.return_type);
-		for (idx_t i = 0; i < aggr.children.size(); ++i) {
-			payload_types.push_back(aggr.children[i]->return_type);
+		for (auto &child : aggr.children) {
+			payload_types.push_back(child->return_type);
+		}
+		if (aggr.filter) {
+			vector<LogicalType> types;
+			vector<vector<Expression *>> bound_refs;
+			BoundAggregateExpression::GetColumnRef(aggr.filter.get(), bound_refs, types);
```

Comment:
This function confuses me a bit; a filter should always only have a single boolean column as input, no? Isn't this just `payload_types.push_back(LogicalType::BOOLEAN)`, or am I missing something?